File: Javascript/Unique-Number-Of-Occurrences.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
const uniqueOccurrences = (arr) => {
  let occurrences = new Map();

  for (num of arr) {
    if (occurrences.has(num)) {
      occurrences.set(num, occurrences.get(num) + 1);
    } else {
      occurrences.set(num, 1);
    }
  }

  return !containsDuplicates(Array.from(occurrences.values()));
};

const containsDuplicates = (nums) => {
  let set = new Set();

  for (number of nums) {
    if (set.has(number)) {
      return true;
    }
    set.add(number);
  }

  return false;
};

File: Go/Unique-Number-Of-Occurrences.go
package main



func uniqueOccurrences(arr []int) bool {
	occurrences := map[int]int{}

	for _, val := range arr {
		occurrences[val]++
	}

	values := []int{}
	for _, value := range occurrences {
		values = append(values, value)
	}

	return !containsDuplicate(values)
}

func containsDuplicate(nums []int) bool {
	set := map[int]int{}

	for _, num := range nums {
		if set[num] != 0 {
			return true
		} else {
			set[num] = 1
		}
	}

	return false
}